packages/opentelemetry-plugin-postgres/test/pg.test.ts

```diff
@@ -0,0 +1,349 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NoopLogger } from '@opentelemetry/core';
+import { NodeTracer } from '@opentelemetry/node';
+import {
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/tracing';
+import { SpanKind, Attributes, TimedEvent, Span } from '@opentelemetry/types';
+import { plugin, PostgresPlugin } from '../src';
+import { AttributeNames } from '../src/enums';
+import * as assert from 'assert';
+import * as pg from 'pg';
+import * as semver from 'semver';
+import * as assertionUtils from './assertionUtils';
+import * as testUtils from './testUtils';
+
+const memoryExporter = new InMemorySpanExporter();
+
+const CONFIG = {
+  user: 'postgres',
+  password: 'test',
+  database: 'postgres',
+  host: '127.0.0.1',
+  port: 5432,
+};
+
+const runCallbackTest = (
+  span: Span,
+  attributes: Attributes,
+  events: TimedEvent[],
+  spansLength = 1,
+  spansIndex = 0
+) => {
+  const spans = memoryExporter.getFinishedSpans();
+  assert.strictEqual(spans.length, spansLength);
+  const pgSpan = spans[spansIndex];
+  assertionUtils.assertSpan(pgSpan, SpanKind.CLIENT, attributes, events);
+  assertionUtils.assertPropagation(pgSpan, span);
+};
+
+describe('pg@7.x', () => {
+  let client: pg.Client;
+  const tracer = new NodeTracer();
+  const logger = new NoopLogger();
+  const testPostgres = process.env.TEST_POSTGRES; // For CI: assumes local postgres db is already available
+  const testPostgresLocally = process.env.TEST_POSTGRES_LOCAL; // For local: spins up local postgres db via docker
+  const shouldTest = testPostgres || testPostgresLocally; // Skips these tests if false (default)
+
+  before(function(ready) {
```

Comment:
Because I'm using `this` as `Mocha.Context` in the `before` fn to skip the tests, its easier to use a `function`. Is there a nicer way to feed mocha my intended `this` with an arrow function?